crates/wasi-common/src/sys/unix/bsd/osfile.rs

```diff
@@ -1,33 +1,18 @@
+use super::super::dir::Dir;
 use std::fs;
-use std::mem::ManuallyDrop;
 use std::ops::{Deref, DerefMut};
 use std::os::unix::prelude::{AsRawFd, RawFd};
 use std::sync::Mutex;
 
-#[derive(Debug)]
-pub(crate) struct DirStream {
-    pub(crate) file: ManuallyDrop<fs::File>,
-    pub(crate) dir_ptr: *mut libc::DIR,
-}
-
-impl Drop for DirStream {
-    fn drop(&mut self) {
-        unsafe { libc::closedir(self.dir_ptr) };
-    }
-}
-
 #[derive(Debug)]
 pub(crate) struct OsFile {
     pub(crate) file: fs::File,
-    pub(crate) dir_stream: Option<Mutex<DirStream>>,
+    pub(crate) dir: Option<Mutex<Dir>>,
```

Comment:
Please add a comment why we're storing it in `OsFile` (or mentioned where the reason is described in more detail)

Reply:
Done.

Reply:
Can you add a reference that BSD nixes require the client to do the caching?

Reply:
Sure thing. It's actually taken directly from the man pages. Let me add that in.